### PR TITLE
+ Add option to emit JSON S-Expressions.

### PR DESCRIPTION
**Pull request by rattrayalex**
Hi there, 

I'm interested in being able to read a Ruby AST from JavaScript. I imagined that others might be interested in cross-language machine-readable access to the Ruby AST as well. 

I have yet to add docs & tests to this PR, but wanted to put it up now as a strawdog for feedback.

I believe that this PR title, once squashed, should pass your commit guidelines; please let me know if there's anything else expected. 

How does this look?

### CODE REVIEW

File: lib/parser/ast/node.rb [hunk 1]
```diff
@@ -34,6 +34,21 @@ def assign_properties(properties)
...
+        children_json = children.map do |child|
+          if child.is_a?(Node)
+            child.to_json
+          else
+            child
+          end
+        end
+
+        [type.to_s, *children_json]
+      end
```
Comment (whitequark) [inline lib/parser/ast/node.rb]: This should go into the `ast` gem, I think.

Comment (rattrayalex) [reply lib/parser/ast/node.rb]: I was thinking the same. I'll submit a PR there. Thanks!

Comment (rattrayalex) [reply lib/parser/ast/node.rb]: (Do you have quick feedback on whether this is roughly the right code? I don't write a ton of ruby)

Comment (whitequark) [reply lib/parser/ast/node.rb]: Code looks OK.

Comment (rattrayalex) [reply lib/parser/ast/node.rb]: Thanks. Posted in https://github.com/whitequark/ast/pull/20, will remove this commit

**Comment by rattrayalex** [comment]
Thanks for the quick feedback! Removed the to_json def. 

This now depends on https://github.com/whitequark/ast/pull/20

**Comment by whitequark** [comment]
Add a dependency on ast 2.4.0. https://rubygems.org/gems/ast

**Comment by rattrayalex** [comment]
Done

**Comment by rattrayalex** [comment]
Thanks!

**Comment by whitequark** [comment]
I still need to review and merge some other PRs before I can make a release.

**Comment by rattrayalex** [comment]
Okay cool. Thanks!

**Pull request merged**